ethereum/core/src/main/java/org/hyperledger/besu/ethereum/vm/MessageFrame.java

```diff
@@ -320,10 +323,27 @@ private MessageFrame(
     this.transactionHash = transactionHash;
     this.revertReason = revertReason;
 
-    this.warmedUpAddresses = new HashSet<>();
-    warmedUpAddresses.add(sender);
-    warmedUpAddresses.add(contract);
-    this.warmedUpStorage = HashMultimap.create();
+    this.warmedUpAddresses = new HashSet<>(accessListWarmAddresses);
+    this.warmedUpAddresses.add(sender);
+    this.warmedUpAddresses.add(contract);
+    this.warmedUpStorage = HashMultimap.create(accessListWarmStorage);
+
+    // the warmed up addresses will always be a superset of the address keys in the warmed up
+    // storage so we can do both warm ups in one pass
+    accessListWarmAddresses
+        .parallelStream()
+        .forEach(
+            address ->
+                Optional.ofNullable(worldState.get(address))
+                    .ifPresent(
+                        account ->
+                            warmedUpStorage
+                                .get(address)
+                                .parallelStream()
+                                .forEach(
+                                    storageKeyBytes ->
+                                        account.getStorageValue(
+                                            UInt256.fromBytes(storageKeyBytes)))));
```

Comment:
@shemnon What are your thoughts on this? Pure rocksdb caching. Should we consider explicit caching for just the access list entries?